### ISSUE TITLE
Regression in 0.57: c_args are not parsed as an array anymore

### ISSUE THREAD

**Issue by avdgrinten** [mesonbuild/meson#8391]
**Describe the bug**
Passing a Meson-formatted array as c_args and similar command line variables breaks in Meson 0.57.

**To Reproduce**
Invoke `meson` with a command line such as: `meson "-Dc_args=['-fno-stack-protector']" "-Dcpp_args=['-fno-stack-protector']" ..`. This causes Meson to invoke the linker as (scroll to the end):
```
c++ -o some_program.so some_program.so.p/input.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,--start-group -Wl,-soname,some_program.so -Wl,--end-group '['"'"'-fno-stack-protector'"'"']'
```
which is obviously broken and results in:
```
c++: error: ['-fno-stack-protector']: No such file or directory
```

Interestingly, compilation of source files still succeeds, only linking is broken.

**Expected behavior**
Meson passes the parameters one by one.

**Comment by jpakkane** [comment]
Does this also happen with 0.57.1?

**Comment by avdgrinten** [comment]
Seems to be fixed in 0.57.1 indeed. Thanks!
